### ISSUE TITLE
[Feature Request] Abort live query

### ISSUE THREAD

**Issue by tobyjaffey** [kairosdb/kairosdb#20]
[See https://groups.google.com/forum/#!topic/kairosdb-group/XMkaPicdaUU]

How about if I could pass a token in my query to identify it, then I can make another HTTP request to abort it.

```
curl -XPOST http://127.0.0.1:8080/api/v1/datapoints/query --data '{"token":"myOpaqueString", ...}'
```

Then, if I want to abort the request I can do something like:

```
curl -XPOST http://127.0.0.1:8080/api/v1/datapoints/query/abort --data '{"token":"myOpaqueString"}'
```

Or,

```
curl -XDELETE http://127.0.0.1:8080/api/v1/datapoints/query/myOpaqueString
```

**Comment by brianhks** [comment]
We were talking about this today and I like the idea of aborting a query.  One simple solution is to abort a query if the client disconnects.  I'm not sure how you can abort an ajax request in a browser other than closing the browser, I'll have to look into it.

Once hurdle with this approach is that determining if a socket is closed in java is a pain the arse.  But I do have a solutions.  A configurable option will turn on query monitoring.  With this option every time a query is made a thread gets started and on a timer sends a single space character to the client.  When the client disconnects this thread will get an exception and notify the query thread to bail out of the query.

Because an extra space doesn't modify the returned json it solves the problem but, if it causes a client grief they can turn it off.  Actually now that I think of it it could be a per request setting that is passed on the query object.

**Comment by lcoulet** [comment]
I like the idea !

But if it's really multithread, there are chances for the single space to
be inserted within the middle of a JSON field and cause problems, so there
has  to be some protections againsat that :)







Cordialement,
Loic COULET

2015-11-19 22:31 GMT+01:00 Brian Hawkins <notifications@github.com>:

**Comment by brianhks** [comment]
Yes when a query finishes it would join with the monitor thread before sending the response.  Definitely wouldn't want an extra space within the json.

**Comment by brianhks** [comment]
So killing queries is easier now with CQL.  I think at the very least I'll add a configurable timeout and a api endpoint where you can list running queries.

**Comment by brianhks** [comment]
Fixed by pull request #609
